### DISCUSSION TITLE
How can i compare ContentItem Part before and update update

### DISCUSSION THREAD

**Discussion by vitalybrandes** [OrchardCMS/OrchardCore#10842] [Q&A]
How can i compare ContentItem Part before and update update
I need to check if there was any update to 1 or more fields in the contantItem Part.

On the ContentPartHandler on Updating or Updated the both ContentItem.UpdatingItem.Content  and  ContantItem.content are the same event if they are not.

**Comment by Skrypt** [comment]
The AuditTrail module has a diff viewer to see what changed on the content item.

**Reply by vitalybrandes** [reply]
@Skrypt 
 I need to create some API call to external system if ContentItem Part was changed and not sure how to do it.
My ContentItem include number of parts and only for one of then i would like to call the API.

**Reply by Skrypt** [reply]
That's called a Webhook which you could create by using the Workflow feature.

**Reply by Skrypt** [reply]
https://github.com/OrchardCMS/OrchardCore/pull/2323

**Reply by Skrypt** [reply]
Without using this custom module. You could use the Published or Created events in the Workflows and have a special task that would do a Diff on that specific field. Would require that you get the actual content item and compare it with the posted form data to see if there is a change on that specific field.

**Reply by vitalybrandes** [reply]
@Skrypt 
Is ContentPartHandler not compatible for this task?

**Reply by Skrypt** [reply]
Depends on the event you want to implement. CreateAsync won't have anything else than the current content item. PublishAsync and UpdateAsync probably will have them, but I remember that one of them is always called I think it's the UpdatingAsync or UpdatedAsync.

**Reply by Skrypt** [reply]
A publish always call an update.
